neuralpp/symbolic/sympy_expression.py

```diff
@@ -312,11 +338,63 @@ def from_sympy_function_and_general_arguments(sympy_function: sympy.Basic, argum
             # see test/sympy_test.py: test_sympy_bug()
             sympy_object = sympy_function(*[sympy_argument.sympy_object for sympy_argument in sympy_arguments],
                                           evaluate=False)
-            return SymPyFunctionApplication(sympy_object, type_dict)
+        elif sympy_function == sympy.Piecewise:
+            sympy_object = sympy_piecewise_from_if_then_else(
+                *[sympy_argument.sympy_object for sympy_argument in sympy_arguments])
         else:
             with sympy.evaluate(False):
                 sympy_object = sympy_function(*[sympy_argument.sympy_object for sympy_argument in sympy_arguments])
-                return SymPyFunctionApplication(sympy_object, type_dict)
+        return SymPyFunctionApplication(sympy_object, type_dict)
+
+
+def sympy_piecewise_from_if_then_else(if_: sympy.Basic, then_: sympy.Basic, else_: sympy.Basic) -> sympy.Piecewise:
+    """ In Piecewise, cond comes after clause. """
+    return sympy.Piecewise((then_, if_), (else_, True))
+
+
+def sympy_piecewise_to_if_then_else(piecewise: sympy.Piecewise) -> Tuple[sympy.Basic, sympy.Basic, sympy.Basic]:
+    return piecewise.args[0][1], piecewise.args[0][0], piecewise.args[1][0]
+
+
+def fold_sympy_piecewise(piecewise_args: List[Tuple[sympy.Basic, sympy.Basic]]) -> sympy.Piecewise:
+    """ `fold` any sympy piecewise to 2-entry piecewise. E.g.,
+    Piecewise((s0, c0), (s1, c1), (s2, True)) will be folded into
+    Piecewise((s0, c0), (Piecewise((s1, c1), (s2, True)), True)
+    Note the last condition must be True, otherwise it cannot be transformed into if-then-else
+    """
+    if len(piecewise_args) < 2:
+        raise TypeError("piecewise is expected to have at least two entries")
+    elif len(piecewise_args) == 2:
+        if not piecewise_args[1][1]:
+            raise TypeError("No else clause: missing default value.")
+        return sympy.Piecewise(*piecewise_args)
+    else:
+        return sympy.Piecewise(piecewise_args[0], (fold_sympy_piecewise(piecewise_args[1:]), True))
+
+
+class SymPyCondFunctionApplication(SymPyFunctionApplicationInterface):
```

Comment:
Please use "Conditional" instead of "Cond" (the idea being that we can always count on identifiers being formed out of full words).